src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -367,135 +376,383 @@ public class SqlMutationEngine : IMutationEngine
+        /// If database policy is defined, the results of the subsequent select statement is used for constructing the response.
+        /// </summary>
+        /// <param name="resultRow">Reuslt of the upsert database operation</param>
+        /// <param name="jsonDocument">Result of the select database operation</param>
+        /// <param name="primaryKeyRoute">Primary key route to be used in the Location Header</param>
+        /// <param name="isReadPermissionConfiguredForRole">Indicates whether read permissions is configured for the role</param>
+        /// <param name="isDatabasePolicyDefinedForReadAction">Indicates whether database policy is configured for read action</param>
+        /// <param name="httpContext">HTTP Context associated with the API request.</param>
+        /// <param name="operationType">Resultant Operation type - Update, Insert, etc.</param>
+        private CreatedResult ConstructCreatedResultResponse(
+            Dictionary<string, object?> resultRow,
+            JsonDocument? jsonDocument,
+            string primaryKeyRoute,
+            bool isReadPermissionConfiguredForRole,
+            bool isDatabasePolicyDefinedForReadAction,
+            EntityActionOperation operationType,
+            HttpContext httpContext)
+        {
+
+            string locationHeaderURL = string.Empty;
+
+            // For PUT and PATCH API requests, the users are aware of the Pks as it is required to be passed in the request URL.
+            // In case of tables with auto-gen PKs, PUT or PATCH will not result in an insert but error out. Seeing that Location Header does not provide users with
+            // any additional information, it is set as an empty string always.
+            // For POST API requests, the primary key route calculated will be an empty string
+            // when the read action for the role does not have access to one or more PKs.
+            // When the computed primaryKeyRoute is non-empty, the location header is calculated.
+            // Location is made up of three parts, the first being constructed from the Host property found in the HttpContext.Request.
+            // The second part being the base route configured in the config file.
+            // The third part is the computed primary key route.
+            if (operationType is EntityActionOperation.Insert && !string.IsNullOrEmpty(primaryKeyRoute))
+            {
+                locationHeaderURL = UriHelper.BuildAbsolute(
+                                        scheme: httpContext.Request.Scheme,
+                                        host: httpContext.Request.Host,
+                                        pathBase: GetBaseRouteFromConfig(),
+                                        path: httpContext.Request.Path);
+
+                locationHeaderURL = locationHeaderURL.EndsWith('/') ? locationHeaderURL + primaryKeyRoute : locationHeaderURL + "/" + primaryKeyRoute;
```

Comment:
Is primaryKeyRoute guaranteed to not have a starting slash / ?

Reply:
Yes